daras_ai_v2/base.py

```diff
@@ -1594,23 +1654,46 @@ def render_extra_waiting_output(self):
     def estimate_run_duration(self) -> int | None:
         pass
 
-    def submit_and_redirect(self):
+    def submit_and_redirect(self) -> typing.NoReturn | None:
         sr = self.on_submit()
         if not sr:
             return
         raise gui.RedirectException(self.app_url(run_id=sr.run_id, uid=sr.uid))
 
+    def publish_and_redirect(self) -> typing.NoReturn | None:
+        assert self.request.user and not self.request.user.is_anonymous
+
+        updated_count = SavedRun.objects.filter(
+            id=self.current_sr.id,
+            uid=self.current_sr_user.uid,
```

Comment:
this filter wont work if you login with a different user?

Reply:
`current_sr_user` is different from the current user (it's taken from `current_sr.uid`)

Reply:
`current_sr.uid` would be clearer here though, thanks!

Reply:
why does this need the uid filter at all? since its filtering on `id`